apps/schematic/api/schematic_api/controllers/utils.py

```diff
@@ -184,3 +193,110 @@ def download_schema_file_as_jsonld(schema_url: str) -> str:
+        minute_buffer (int): All files created this amount of time or older will be deleted
+    """
+    # try clearing the cache
+    # scan a directory and check size of files
+    if os.path.exists(store.root_synapse_cache):
+        maximum_storage_allowed_cache_bytes = maximum_storage_allowed_cache_gb * (
+            1024**3
+        )
+        dir_size_bytes = check_synapse_cache_size(directory=store.root_synapse_cache)
+        # if 1 GB has already been taken, purge cache before 15 min
+        if dir_size_bytes >= maximum_storage_allowed_cache_bytes:
+            minutes_earlier = calculate_datetime(minute_buffer)
+            num_of_deleted_files = store.syn.cache.purge(before_date=minutes_earlier)
+            LOGGER.info(
+                f"{num_of_deleted_files}  files have been deleted from {store.root_synapse_cache}"
+            )
+        else:
+            # on AWS, OS takes around 14-17% of our ephemeral storage (20GiB)
+            # instead of guessing how much space that we left, print out .synapseCache here
+            LOGGER.info(f"the total size of .synapseCache is: {dir_size_bytes} bytes")
+
+
+def check_synapse_cache_size(
+    directory: str = "/root/.synapseCache",
+) -> float:
+    """use du --sh command to calculate size of .synapseCache.
+
+    Args:
+        directory (str, optional): .synapseCache directory. Defaults to '/root/.synapseCache'
+
+    Returns:
+        float: returns size of .synapsecache directory in bytes
+    """
+    # Note: this command might fail on windows user.
+    # But since this command is primarily for running on AWS, it is fine.
+    command = ["du", "-sh", directory]
+    output = subprocess.run(command, capture_output=True, check=False).stdout.decode(
+        "utf-8"
+    )
+
+    # Parsing the output to extract the directory size
+    size = output.split("\t")[0]
+    return calculate_byte_size(size)
+
+
+def calculate_byte_size(size_string: str) -> int:
+    """
+    Calculates the size in bytes of a size returned from the "du" command
+
+    Args:
+        size_string (str):
+          The input must be a string such as 4B, or 1.2K.
+          Sizes up to GB allowed.
+
+    Raises:
+        ValueError: When the input doesn't match the allowed paterns
+
+    Returns:
+        int: The size in bytes
+    """
+    size_dict: dict[str, int] = {"B": 0, "K": 1, "M": 2, "G": 3}
+
+    size_letter_string = "".join(size_dict.keys())
+    int_size_match = re.match(f"^[0-9]+[{size_letter_string}]$", size_string)
+    float_size_match = re.match(f"^[0-9]+\.[0-9]+[{size_letter_string}]$", size_string)
+    if not (int_size_match or float_size_match):
+        LOGGER.error("Cannot recognize the file size unit")
+        raise ValueError("The size string doesn't match the allowed type:", size_string)
+
+    size_letter = size_string[-1]
+    size = float(size_string[:-1])
+    multiple = 1024 ** size_dict[size_letter]
+    byte_size: int = ceil(size * multiple)
```

Comment:
Yeah sorry for the previous comment. I just realized that this will be `int`
But do we do inline typing like this? I think having int as return type is sufficient?